auto_submit/lib/validations/revert.dart

```diff
@@ -100,4 +129,73 @@ class Revert extends Validation {
     List<String> linkSplit = link.split('#');
     return int.parse(linkSplit.elementAt(1));
   }
+
+  /// Wait for the required checks to complete.
+  Future<bool> waitForRequiredChecks({
+    required GithubService githubService,
+    required github.RepositorySlug slug,
+    required String sha,
+    required List<String> checkNames,
+  }) async {
+    List<github.CheckRun> targetCheckRuns = [];
+    for (var element in checkNames) {
+      targetCheckRuns.addAll(
+        await githubService.getCheckRunsFiltered(
+          slug: slug,
+          ref: sha,
+          checkName: element,
+        ),
+      );
+    }
+
+    bool checksCompleted = true;
+
+    try {
+      for (github.CheckRun checkRun in targetCheckRuns) {
+        await _waitForRequiredCheckRunsWithRetries(
+          () async {
+            await _verifyCheckRunCompleted(
+              slug,
+              githubService,
+              checkRun,
+            );
+          },
+          retryOptions,
+        );
+      }
+    } catch (e) {
+      log.warning('Required check has not completed in time. ${e.toString()}');
+      checksCompleted = false;
+    }
+
+    return checksCompleted;
+  }
+}
+
+/// Function signature that will be executed with retries.
+typedef RetryHandler = Function();
+
+/// Runs the internal verify checks runs function with retries.
+Future<void> _waitForRequiredCheckRunsWithRetries(RetryHandler retryHandler, RetryOptions retryOptions) {
```

Comment:
This is simple enough that you may want to inline it

Reply:
Done.